flake.nix

```diff
@@ -0,0 +1,94 @@
+{
+  description = "Cross-platform visual development tool and SCPI instrument controller";
+
+  inputs = {
+    nixpkgs.url = "nixpkgs/nixos-21.05";
+    flake-utils.url = "github:numtide/flake-utils";
+    nix-filter.url = "github:numtide/nix-filter";
+    npmlock2nix = {
+      url = "github:ilkecan/npmlock2nix/";
+      flake = false;
+    };
+    nix-utils.url = "git+https://git.sr.ht/~ilkecan/nix-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils, npmlock2nix, ... }@inputs:
+    let
+      inherit (builtins)
+        attrNames
+        attrValues
+      ;
+      inherit (nixpkgs.lib)
+        getAttrs
+      ;
+      inherit (flake-utils.lib)
+        defaultSystems
+        eachSystem
+      ;
+      nix-filter = inputs.nix-filter.lib;
+      nix-utils = inputs.nix-utils.lib;
+      inherit (nix-utils)
+        createOverlays
+        getUnstableVersion
+      ;
+
+      supportedSystems = defaultSystems;
```

Comment:
While it is mentioned in the README that 64-bit OS is required for installation, the flake currently produces outputs for the default systems supported by Nixpkgs, which also includes some 32-bit architectures. As far as I can tell Electron also supports 32 bits systems so I am not sure where does this 64-bit requirement comes from but if you want I can filter out the 32-bit systems in here.

Reply:
I think there is no real technical reason why we don't support 32 bits. It is mentioned in README because we only build ant test 64 bits executables.